lib/paths.ts

```diff
@@ -0,0 +1,12 @@
+import { useRouter } from "next/router";
+
+import { pagesPath } from "@/lib/$path";
+
+import { DEFAULT_CHANNEL,DEFAULT_LOCALE } from "./regions";
+
+export const usePaths = () => {
+  const { query } = useRouter();
+  const channel = query.channel?.toString() || DEFAULT_CHANNEL.slug;
+  const locale = query.locale?.toString() || DEFAULT_LOCALE;
+  return { paths: pagesPath._channel(channel)._locale(locale) };
```

Comment:
Why is it artificially nested inside an object? Wouldn't this usage make more sense?
```
const paths = usePaths();
```